tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -923,7 +927,8 @@ def read_file(path: str) -> str:
                         "publisher": "[parameters('publisher')]"
                     }
 
-        AddNetworkSecurityGroup().update(template, is_lisa_template=False)
+        if self._allow_ssh != '':
```

Comment:
This give me impression this value can be empty, and we let go without failing the tests? Eventually test fails with ssh issue since rule was not added because ip empty. I think we should fail fast instead tests fails with ssh rule

Reply:
yes, it can be empty

(from runbook.yml)
```
  #
  # Takes an IP address as value; if not empty, it adds a Network Security Rule allowing SSH access from the specified IP address to any test VMs created by the runbook execution.
  #
  - name: allow_ssh
    value: ""
    is_case_visible: true
    
```

from the PR description

`
This rule, however, now needs to be added explicitly during test runs using the new "allow_ssh" runbook parameter.
`

Reply:
My point is tests may fail with ssh issue if user not providing as runbook parameter, why don't we let user to know by raising exception that need this parameter when it's empty

Reply:
empty is not an error.

short tests execute without issue. longer runs may hit issues.

Reply:
automation runs always set this parameter.

the intention here is to force developers to think about what they are doing (adding a rule that can cause problems). developers many times forget to delete their test VMs and we get security warnings on the test subscription. if the developer needs to add the rule, I want them to be aware of what they are doing.

if the developer doesn't need to add a rule because they are just executing a short test, I don't want to force them to add a rule (i.e. I do not want to report that as an error)

Reply:
makes sense